touchforms/backend/persistence.py

```diff
@@ -24,32 +29,30 @@ def restore(sess_id, factory, override_state=None):
         state.update(override_state)
     return factory(**state)
 
-# TODO integrate with real caching framework (django ideally)
+
 def cache_set(key, value, timeout):
-    with open(cache_path(key), 'w') as f:
-        f.write(json.dumps(value).encode('utf8'))
+    tf_db = get_map_db()
+    sess_map = tf_db.getTreeMap("touchforms-map")
+    sess_map.put(key, value)
+    tf_db.commit()
+
 
-# TODO integrate with real caching framework (django ideally)
 def cache_get(key):
-    try:
-        with open(cache_path(key)) as f:
-            return json.loads(f.read().decode('utf8'))
-    except IOError:
-        raise KeyError
-    except JSONDecodeError:
-        raise EmptyCacheFileException(_(
-            u"Unfortunately an error has occurred on the server and your form cannot be saved. "
-            u"Please take note of the questions you have filled out so far, then refresh this page and enter them again. "
-            u"If this problem persists, please report an issue."
-        ))
+    tf_db = get_map_db()
+    sess_map = tf_db.getTreeMap("touchforms-map")
+    return sess_map.get(key)
 
 
 def cache_del(key):
-    raise RuntimeError('not implemented')
-
-def cache_path(key):
-    # todo: make this use something other than the filesystem
-    persistence_dir = settings.PERSISTENCE_DIRECTORY or tempfile.gettempdir()
-    if not os.path.exists(persistence_dir):
-        os.makedirs(persistence_dir)
-    return os.path.join(persistence_dir, 'tfsess-%s' % key)
+    tf_db = get_map_db()
+    sess_map = tf_db.getTreeMap("touchforms-map")
+    sess_map.remove(key)
+
+
+def get_map_db():
+    temp_file = (settings.PERSISTENCE_DIRECTORY or tempfile.gettempdir()) + "/tf-map-db"
+    f = File(temp_file)
+    if not f.exists():
+        f = File(temp_file)
+    db = DBMaker.newFileDB(f).closeOnJvmShutdown().make()
```

Comment:
took a quick look at `MapDb` and it doesn't seem like it will do the trick.

looks like you could assume postgres and connect directly using https://jdbc.postgresql.org/

google also turned up http://ormlite.com/ as a possible alternative.